### DISCUSSION TITLE
Endoutpost ruin walls fail to render behind submarine

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14595] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In the end outpost, you can see the ruin walls rendering in front of the wreck walls

I'm assuming `drawbehindsubs="true"` broke?

https://github.com/FakeFishGames/Barotrauma/blob/0e8fb6569d2810e2f8ad5fb17b4bba546cc5739a/Barotrauma/BarotraumaShared/SharedSource/Map/Outposts/OutpostGenerator.cs#L432-L439

![image](https://github.com/user-attachments/assets/84727cf9-73ad-4461-822b-84ce2151a184)

### Reproduction steps

.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by NotWendy** [comment]
The wreck is built into the level. There's no difference between it and the outpost itself. The way to fix it is to simply increase the depth of the temple's wall by a bit.

**Reply by ubertpendragon** [reply]
True, both are set at 0.980, I don't recall the ruin wall appearing at the same depth as the sub though

**Reply by NotWendy** [reply]
The whole thing renders behind our submarine because of drawbehindsubs="true", I assume. But the wreck isn't treated as a submarine, because it's not spawned by the level generator from a separate .sub file, but rather pre-placed.
And with matching depths, it's a coinflip for whether the wreck or the temple's background wall will get rendered in front when the level loads.

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in then next Unstable build/patch. Closing discussion.
